### PR TITLE
Change NOSTR client to multi connections

### PR DESCRIPTION
**Pull request by DavidGershony**

### CODE REVIEW

File: src/Angor/Client/Pages/Invest.razor [hunk 1]
```diff
@@ -472,6 +472,9 @@
 
     private async Task HandleSignatureReceivedAsync(string? nostrPrivateKeyHex, string _)
     {
+        if (recoverySigs?.Signatures.Any() ?? false) //multiple relays for the same message
```
Comment (dangershony) [inline src/Angor/Client/Pages/Invest.razor:475]: I wonder if we should compare sigs (probably no need) we filter using pub key which is unique per user per project right?

File: src/Angor/Client/Pages/Settings.razor [hunk 2]
```diff
@@ -96,7 +97,8 @@
                 {
                     <tr>
                         <td><a href="@relay.Url" target="_blank">@relay.Url</a></td>
-                            <td style="color: @(relay.Status == UrlStatus.Online ? "green" : relay.Status == UrlStatus.NotReady ? "yellow" :  "red");">@relay.Status.ToString()</td>
+                        <td>@relay.Name</td>
```
Comment (dangershony) [inline src/Angor/Client/Pages/Settings.razor:100]: A relay has a name? is it assigned or fetched form the relay?

File: src/Angor/Client/Pages/Signatures.razor [hunk 3]
```diff
@@ -172,43 +148,57 @@
...
                     pendingSignature.AmountToInvest = investorTrx.Outputs.AsIndexedOutputs().Skip(2).Take(investorTrx.Outputs.Count - 3) //Todo get the actual outputs with taproot type
                         .Sum(_ => _.TxOut.Value);
-                    
-                    StateHasChanged();
                 }
                 catch (Exception e)
                 {
                     Console.WriteLine(e);
-                    Console.WriteLine(pendingSignature.TransactionHex);
+                    _Logger.LogDebug(pendingSignature.TransactionHex);
```
Comment (dangershony) [inline src/Angor/Client/Pages/Signatures.razor:155]: probably not logging the correct thing here?

File: src/Angor/Shared/DerivationOperations.cs [hunk 4]
```diff
@@ -216,6 +216,22 @@ public Key DeriveProjectNostrPrivateKey(WalletWords walletWords, int index)
 
         return extKey.PrivateKey;
     }
+    
+    public async Task<Key> DeriveProjectNostrPrivateKeyAsync(WalletWords walletWords, int index)
+    {
+        // founder key is derived from the path m/5'
+        ExtKey extendedKey = GetExtendedKey(walletWords);
+
+        var path = $"m/44'/1237'/{index}/0/0";
```
Comment (dangershony) [inline src/Angor/Shared/DerivationOperations.cs:225]: why 1237?

File: src/Angor/Shared/Models/NostrRelayInfo.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+using System.Text.Json.Serialization;
+
+namespace Angor.Shared.Models;
+
+public class NostrRelayInfo
+{
+    [JsonPropertyName("name")]
+    public string Name { get; set; }
```
Comment (dangershony) [inline src/Angor/Shared/Models/NostrRelayInfo.cs:8]: ah ok yes we fetch this data from the relay?

**Review by dangershony — APPROVED** [review]
Looks ok to me

File: src/Angor/Client/Pages/Invest.razor (hunk 1 above)
Comment (DavidGershony) [reply src/Angor/Client/Pages/Invest.razor:475]: The sigs will be different for the same user and project as well

File: src/Angor/Client/Pages/Settings.razor (hunk 2 above)
Comment (DavidGershony) [reply src/Angor/Client/Pages/Settings.razor:100]: Yes, I have updated the code to do that

File: src/Angor/Client/Pages/Signatures.razor (hunk 3 above)
Comment (DavidGershony) [reply src/Angor/Client/Pages/Signatures.razor:155]: It's only for debugging, I will remove it..

File: src/Angor/Shared/DerivationOperations.cs (hunk 4 above)
Comment (DavidGershony) [reply src/Angor/Shared/DerivationOperations.cs:225]: NOSTR protocol

**Pull request merged**